src/gui/StyleSpec.h

```diff
@@ -0,0 +1,139 @@
+/*
+Minetest
+Copyright (C) 2019 rubenwardy
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "irrlichttypes_extrabloated.h"
+#include <array>
+
+#pragma once
+
+class StyleSpec
+{
+public:
+	enum Property
+	{
+		TEXTCOLOR,
+		BGCOLOR,
+		NOCLIP,
+		BORDER,
+		BGIMG,
+		BGIMG_PRESSED,
+		ALPHA,
+		NUM_PROPERTIES,
+		NONE
+	};
+
+private:
+	std::array<bool, NUM_PROPERTIES> property_set;
+	std::array<std::string, NUM_PROPERTIES> properties;
+
+public:
+	static Property GetPropertyByName(const std::string &name)
+	{
+		if (name == "textcolor") {
+			return TEXTCOLOR;
+		} else if (name == "bgcolor") {
+			return BGCOLOR;
+		} else if (name == "noclip") {
+			return NOCLIP;
+		} else if (name == "border") {
+			return BORDER;
+		} else if (name == "bgimg") {
+			return BGIMG;
+		} else if (name == "bgimg_pressed") {
+			return BGIMG_PRESSED;
+		} else if (name == "alpha") {
+			return ALPHA;
+		} else {
+			return NONE;
+		}
+	}
+
+	std::string get(Property prop, std::string def) const
+	{
+		const auto &val = properties[prop];
+		if (val.empty()) {
```

Comment:
`return properties[prop].empty() ? def : val`
Or do you plan to add more to this function?